src/modules/common/components/customRoute/index.js

```diff
@@ -30,19 +31,30 @@ const CustomRoute = ({
   const wallet = useSelector(state => selectActiveTokenAccount(state));
   const token = useSelector(state => state.token);
   const isNetworkSet = useSelector(checkNetwork);
-  const isAuthenticated = !!wallet.summary;
+  const [account] = useCurrentAccount();
+  const isAuthenticated = Object.keys(account).length > 0;
   const { search = '' } = history.location;
+  const { accounts } = useAccounts();
+  const hasAccounts = accounts.length > 0;
```

Comment:
if the logic is not reused, its better to use them directly to avoid unnecessary allocation